### PR TITLE
Add Ethereum support

### LINKED ISSUE #32: Add Ethereum support with yui-ibc-solidity

### PR DESCRIPTION
**Pull request by bluele**
resolves #32 

This PR aims to support the ethereum with yui-ibc-solidity.

This includes the following:
- Implement Chain for ethereum 
- Introduce the global codec instead of using the codec per chain

This does not include the following:
- Prover for IBFT2 client is not supported

### CODE REVIEW

File: chains/ethereum/chain.go [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+	"github.com/hyperledger-labs/yui-ibc-solidity/pkg/wallet"
+	"github.com/hyperledger-labs/yui-relayer/core"
+)
+
+type Chain struct {
+	config ChainConfig
+
+	pathEnd        *core.PathEnd
+	homePath       string
+	encodingConfig params.EncodingConfig
```
Comment (siburu) [inline chains/ethereum/chain.go:33]: It is a minor detail, but `encodingConfig` is removed from other chains and I think is not needed for `ethereum` either.

Comment (bluele) [reply chains/ethereum/chain.go]: You are right. This is fixed at https://github.com/hyperledger-labs/yui-relayer/pull/37/commits/30de154c36254732ed7d6e9193943691c4cddf73

**Comment by bluele** [comment]
@siburu I've applied your suggestion, so can you review them again?

**Review by siburu — APPROVED** [review]
👌

**Pull request merged**